src/lib/ares_dns_record.c

```diff
@@ -1314,3 +1314,79 @@ ares_bool_t ares_dns_has_opt_rr(const ares_dns_record_t *rec)
   }
   return ARES_FALSE;
 }
+
+/* Construct a DNS record for a name with given class and type. Used internally
+ * by ares_search() and ares_create_query().
+ */
+ares_status_t ares_dns_record_create_query(ares_dns_record_t **dnsrec,
```

Comment:
Lets use the proper types here even if any public calling function does not (because they predate the types):
 * `ares_dns_class_t dnsclass`
 * `ares_dns_rec_type_t type`
 * `int rd` ->  `ares_dns_flags_t flags` (`flags & ARES_FLAG_RD`)
 * `size_t max_udp_size`

Reply:
actually I think `unsigned short` here is still not optimal.  Since this is an internal function, and not a public one, using the `ares_dns_flags_t` type here is more clear.  In a public function we can't do that because of C++ interop, but in C its ok to use an enum as flags without needing to use an integer type.

Reply:
Ah OK, that makes sense about why we need different types for public vs. private functions. I have switched this to `ares_dns_flags_t` and am casting to `unsigned short` before calling into the public `ares_dns_record_create()`.